cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -106,6 +106,8 @@ void SparsePoseGraph::AddScan(
   });
   trajectory_connectivity_.Add(submaps);
 
+  // if the last (i.e. latest) insertion_submap is not in the index
+  // it means we have started a new submap...
```

Comment:
Rather than repeating the condition, move this comment inside the if. This way, comments and code cannot diverge as easily. For example:
```
    // We have started a new submap.
```